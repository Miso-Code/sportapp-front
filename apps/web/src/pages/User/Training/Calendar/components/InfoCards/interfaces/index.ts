import { ReactNode } from 'react'

export enum StateTypes {
	BACK = 'default',
	TODAY = 'primary',
	NEXT = 'warning'
}

export interface Props {
	titleComponent: string | ReactNode
	description: string | ReactNode
	state: { type: `${keyof typeof StateTypes}`; text: string }
	date: string
	showChip?: boolean
	onClick: () => void
}
