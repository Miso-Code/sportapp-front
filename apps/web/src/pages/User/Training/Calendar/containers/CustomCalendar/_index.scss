@import 'react-big-calendar/lib/sass/styles';

.training-calendar-component {
	.rbc-toolbar-label {
		@apply uppercase font-bold text-lg;
	}

	.rbc-header {
		@apply uppercase font-bold text-sm;
	}

	.rbc-toolbar button:active,
	.rbc-toolbar button.rbc-active {
		@apply bg-[#425E91] text-white;
	}

	.rbc-toolbar {
		button:hover {
			@apply bg-[#425E91] text-white;
		}

		button:focus {
			@apply bg-[#425E91] text-white;
		}
	}

	.event-custom-type {
		&-training {
			.rbc-event {
				@apply bg-[#425E91] text-white;
			}
		}

		&-event {
			.rbc-event {
				@apply bg-[#F9A825] text-white;
			}
		}

		&-session {
			.rbc-event {
				@apply bg-[#5e339a] text-white;
			}
		}
	}
}
