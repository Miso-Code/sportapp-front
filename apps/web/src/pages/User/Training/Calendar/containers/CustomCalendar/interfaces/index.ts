import { ReactNode } from 'react'
import { Event, EventWrapperProps } from 'react-big-calendar'

export interface Props {
	events: CustomEvent[]
	onSelectEvent?: (event: CustomEvent) => void
}

export enum EventTypes {
	EVENT = 'event',
	TRAINING = 'training',
	SESSION = 'session'
}

export interface CustomEvent extends Event {
	id?: string
	custom_class?: string
	description?: string
	type?: `${EventTypes}`
	warm_up?: number
	cardio?: number
	strength?: number
	cool_down?: number
	location_latitude?: number
	location_longitude?: number
	capacity?: number
}

export interface EventWrapperCustomProps
	extends EventWrapperProps<CustomEvent> {
	children?: ReactNode
}
