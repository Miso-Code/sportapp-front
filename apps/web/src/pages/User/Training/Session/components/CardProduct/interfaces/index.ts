export interface Props {
	image: string
	title: string
	description: string
	onClick?: () => void
}
