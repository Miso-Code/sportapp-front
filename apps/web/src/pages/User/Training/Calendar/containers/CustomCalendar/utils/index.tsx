import { EventWrapperCustomProps } from '../interfaces'

export const components = {
	eventWrapper: (props: EventWrapperCustomProps) => {
		const { event } = props
		return (
			<div
				className={`event-wrapper event-custom-type-${event.type} ${event.custom_class ?? ''}`}>
				{props.children}
			</div>
		)
	}
}
