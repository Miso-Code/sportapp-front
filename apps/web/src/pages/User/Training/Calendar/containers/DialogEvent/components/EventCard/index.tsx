import { Box, Card, CardContent, Typography } from '@mui/material'
import { format } from 'date-fns'
import { es } from 'date-fns/locale'
import { CustomEvent } from '../../../CustomCalendar/interfaces'

export default function EventCard({
	selectedValue
}: {
	selectedValue: CustomEvent
}) {
	return (
		<Card>
			<CardContent>
				<Box
					sx={{
						display: 'flex',
						justifyContent: 'space-between',
						pb: 2
					}}>
					<Typography variant='body1' color='GrayText'>
						{format(
							selectedValue.start as Date,
							'dd MMMM yyyy hh:mm a',
							{
								locale: es
							}
						)}
					</Typography>
				</Box>
				<Typography gutterBottom variant='h6'>
					{selectedValue.title}
				</Typography>
				<Typography variant='body2' color='GrayText' component='div'>
					{selectedValue.description}
				</Typography>
			</CardContent>
		</Card>
	)
}
