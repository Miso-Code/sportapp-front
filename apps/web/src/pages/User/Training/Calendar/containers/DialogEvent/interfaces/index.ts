import { CustomEvent } from '../../../containers/CustomCalendar/interfaces'

export interface Props {
	open: boolean
	selectedValue: CustomEvent
	onClose: () => void
}
