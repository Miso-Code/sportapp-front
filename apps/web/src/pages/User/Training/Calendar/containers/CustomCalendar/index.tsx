import format from 'date-fns/format'
import getDay from 'date-fns/getDay'
import enUS from 'date-fns/locale/en-US'
import es from 'date-fns/locale/es'
import parse from 'date-fns/parse'
import startOfWeek from 'date-fns/startOfWeek'
import { Calendar, dateFnsLocalizer } from 'react-big-calendar'
import { useTranslation } from 'react-i18next'
import { Props } from './interfaces'
import { components } from './utils'

import './_index.scss'

const locales = {
	es: es,
	en: enUS
}

export default function CustomCalendar({ events, onSelectEvent }: Props) {
	const { t, i18n } = useTranslation()

	const localizer = dateFnsLocalizer({
		format,
		parse,
		startOfWeek,
		getDay,
		locales
	})

	return (
		<Calendar
			components={components}
			localizer={localizer}
			events={events}
			culture={i18n.language}
			startAccessor='start'
			endAccessor='end'
			messages={{
				allDay: t('training.allDay'),
				previous: t('training.previous'),
				next: t('training.next'),
				today: t('training.today'),
				month: t('training.month'),
				week: t('training.week'),
				day: t('training.day'),
				date: t('training.date'),
				time: t('training.hour'),
				event: t('training.event'),
				showMore: (total) => t('training.showMore', { total }),
				agenda: t('training.schedule'),
				noEventsInRange: t('training.nextEventsEmpty'),
				tomorrow: t('training.tomorrow'),
				yesterday: t('training.yesterday'),
				work_week: t('training.workWeek')
			}}
			onSelectEvent={onSelectEvent}
			className='training-calendar-component'
			style={{ maxHeight: 'calc(100vh - 200px)' }}
		/>
	)
}
