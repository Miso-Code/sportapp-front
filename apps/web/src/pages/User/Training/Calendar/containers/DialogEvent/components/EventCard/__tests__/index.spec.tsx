import EventCard from '..'
import { render, RenderResult } from '@testing-library/react'
import { CustomEvent, EventTypes } from '../../../../CustomCalendar/interfaces'

describe('EventCard', () => {
	let wrapper: RenderResult

	beforeEach(() => {
		const selectedValue: CustomEvent = {
			type: EventTypes.EVENT,
			title: 'title',
			id: 'id',
			start: new Date(2024, 9, 1, 0, 0, 0, 0),
			end: new Date(2024, 9, 1, 0, 0, 0, 0),
			description: 'description'
		}
		wrapper = render(<EventCard selectedValue={selectedValue} />)
	})

	it('should render', () => {
		expect(wrapper.container).toMatchSnapshot()
	})
})
