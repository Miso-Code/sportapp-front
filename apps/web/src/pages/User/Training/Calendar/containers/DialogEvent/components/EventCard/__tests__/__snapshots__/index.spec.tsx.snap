// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventCard should render 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <div
        class="MuiBox-root css-1jyfzin"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1m8eymn-MuiTypography-root"
        >
          01 octubre 2024 12:00 AM
        </p>
      </div>
      <h6
        class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-18k87ye-MuiTypography-root"
      >
        title
      </h6>
      <div
        class="MuiTypography-root MuiTypography-body2 css-x6zrxh-MuiTypography-root"
      >
        description
      </div>
    </div>
  </div>
</div>
`;
